### PR TITLE
Dev renpycompat

### PR DESCRIPTION
**Pull request by madeddy**
Separation of specials into own module

### CODE REVIEW

**Review by CensoredUsername — COMMENTED** [review]
"codegen" should indeed be removed. But `magic` and `translate` are still submodules of `decompiler` so those ought to stay.

File: decompiler/renpycompat.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        if isinstance(state, tuple):
+            self.update(state[0].keys())
+        else:
+            self.update(state)
+
+
+CLASS_FACTORY = magic.FakeClassFactory(SPECIAL_CLASSES, magic.FakeStrict)
+
+
+def sf_unpickle_str(buffer: bytes):
```
Comment (CensoredUsername) [inline decompiler/renpycompat.py:111]: For these functions, I'd keep more in line with python's naming convention. so just call them something like `pickle_load`, `pickle_loads`, `pickle_dump`, `pickle_dumps`.

Comment (madeddy) [reply decompiler/renpycompat.py:111]: Yeah, i did try to find something. Problem we have all four as "safe" variants and the two load(s).
How about now?

**Comment by madeddy** [comment]
> `magic` and `translate` are still submodules

OK. I misunderstood the role of the exports in __init__ somehow. I added renpycompat. Is this correct?

**Comment by CensoredUsername** [comment]
>I added renpycompat. Is this correct?

Good one, yep that is correct.

I just spotted this btw:
># if type hints: which one would be output file? bytesIO or bytes?

the correct one is typing.BinaryIO for this, which covers both.

so 
```python
from typing import BinaryIO

def random_func(outfile: BinaryIO):
    pass
```

**Comment by CensoredUsername** [comment]
Anyway, looks good, if you don't mind can you squash it into a single commit? I can also do that at merge time but then it messes with the commit authorship I think.

**Comment by madeddy** [comment]
> ...can you squash it into a single commit

I did and something happened in Codium and the sync did error so i needed to do a reset. Should be OK now, i checked again but better look also.

Got the strange feel i forgot something. But what. 😖

**Comment by CensoredUsername** [comment]
Looked fine to me so I merged it, and I also propagated support for it in un.rpyc on `dev`.

**Pull request merged**